ruby/rspec/style_guide.md

```diff
@@ -0,0 +1,645 @@
+      expect(subject.summary).to be_nil
+    end
+  end
+end
+```
+
+#### Good Example
+
+```ruby
+subject { double('article') }
+
+describe '#summary' do
+  context 'when summary is not present' do
+    # This is no longer stubbing all of the functionality, and will
+    # actually test the objects handling of the methods return value.
+    it 'returns nil' do
+      allow(subject).to receive(:summary).and_return(nil)
+      expect(subject.summary).to be_nil
+    end
+  end
+end
+```
+
+### Dealing with Time
+
+Always use [Timecop](https://github.com/travisjeffery/timecop) instead of
+stubbing anything on Time or Date.
+
+#### Bad Example
+
+```ruby
+it 'offsets the time 2 days into the future' do
+  current_time = Time.now
+  allow(Time).to receive(:now).and_return(current_time)
+  expect(subject.get_offset_time).to be_the_same_time_as (current_time + 2.days)
+end
+```
+
+#### Good Example
+
+```ruby
+it 'offsets the time 2 days into the future' do
+  Timecop.freeze(Time.now) do
+    expect(subject.get_offset_time).to eq 2.days.from_now
+  end
+end
+```
+
+
+### `let` blocks
+
+Use `let` blocks instead of `before(:each)` blocks to create data for the spec
+examples. `let` blocks get lazily evaluated. It also removes the instance
+variables from the test suite (which don't look as nice as local variables).
+
+These should primarily be used when you have duplication among a number of
+`it` blocks within a `context` but not all of them. Be careful with overuse of
+`let` as it makes the test suite much more difficult to read.
+
+```ruby
+# use this:
+let(:article) { create(:article) }
+
+# ... instead of this:
+before { @article = create(:article) }
+```
+
+### `subject`
+
+Use `subject` when possible
+
+```ruby
+describe Article do
+  subject { create(:article) }
+
+  it 'is not published on creation' do
+    expect(subject).not_to be_published
+  end
+end
+```
+
+### Magic Matchers
+
+Use RSpec's 'magical matcher' methods when possible. For instance, a class
+with the method `published?` should be tested with the following:
+
+```ruby
+it 'is published' do
+  # actually tests subject.published? == true
+  expect(subject).to be_published
+end
+```
+
+### Incidental State
+
+Avoid incidental state as much as possible.
+
+#### Bad Example
+
+```ruby
+it 'publishes the article' do
+  article.publish
+
+  # Creating another shared Article test object above would cause this
+  # test to break
+  expect(Article.count).to eq(2)
+end
+```
+
+#### Good Example
+
+```ruby
+it 'publishes the article' do
+  expect { article.publish }.to change(Article, :count).by(1)
+end
+```
+
+### DRY
+
+Be careful not to focus on being 'DRY' by moving repeated expectations into a shared environment too early, as this can lead to brittle tests that rely too much on one other.
+
+It general it is best to start with doing everything directly in your `it`
+blocks even if it is duplication and then refactor your tests after you have
```

Comment:
The sentence is not necessarily bad, but it could be improved....
Maybe:
"...refactor your tests once they are working, to be a little more DRY"

Reply:
Great job, by the way :)